From acb9438f64104ecd307a64d4915ecbfc4ac6852b Mon Sep 17 00:00:00 2001
From: Nicolas Chevalier <nchevali@qti.qualcomm.com>
Date: Fri, 27 Sep 2024 16:01:55 +0200
Subject: [PATCH] shader: Avoid requiring GL_EXT_texture_shadow_lod if not
 supported.

This patch ensures that the 'GL_EXT_texture_shadow_lod' extension is
not required or appended in the shader header when it is not supported
by the host. This prevents potential shader compilation issues on
hosts that lack support for this extension, while maintaining correct
behavior for features around GLSL sampling that do not require this
extension.

Signed-off-by: Nicolas Chevalier <nchevali@qti.qualcomm.com>
---
 src/vrend_shader.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/vrend_shader.c b/src/vrend_shader.c
index bf3873fa..bcd4cc54 100644
--- a/src/vrend_shader.c
+++ b/src/vrend_shader.c
@@ -3344,7 +3344,7 @@ static void translate_tex(struct dump_ctx *ctx,
    bool has_bias = strbuf_get_len (&bias_buf) != 0;
    bool has_offset = strbuf_get_len (&offset_buf) != 0;
    // EXT_texture_shadow_lod defines a few more functions handling bias
-   if (has_bias &&
+   if (has_bias && ctx->cfg->has_texture_shadow_lod &&
        (inst->Texture.Texture == TGSI_TEXTURE_SHADOW2D_ARRAY ||
         inst->Texture.Texture == TGSI_TEXTURE_SHADOWCUBE ||
         inst->Texture.Texture == TGSI_TEXTURE_SHADOWCUBE_ARRAY))
@@ -3352,6 +3352,7 @@ static void translate_tex(struct dump_ctx *ctx,
 
    // EXT_texture_shadow_lod also adds the missing textureOffset for 2DArrayShadow in GLES
    if ((has_bias || has_offset) && ctx->cfg->use_gles &&
+       ctx->cfg->has_texture_shadow_lod &&
        (inst->Texture.Texture == TGSI_TEXTURE_SHADOW1D_ARRAY ||
         inst->Texture.Texture == TGSI_TEXTURE_SHADOW2D_ARRAY))
       ctx->shader_req_bits |= SHADER_REQ_TEXTURE_SHADOW_LOD;
-- 
2.25.1

